app/models/image_set.rb

```diff
@@ -20,6 +20,18 @@ class ImageSet < ActiveRecord::Base
 
   validate :valid_tags
 
+  accepts_nested_attributes_for :images
+
+  validates_associated :images
+  validate :main_image_in_image_set
+  validate :main_image_is_public
+
+  # make sure lion_id is nil or pointing to a real lion
+  validates :lion, presence: true, if: "lion_id.present?"
```

Comment:
was not aware you can pass a string instead of a proc to `:if`. I think rails has more semantic ways to do this, but if it works it works.